pep-0639.rst

```diff
@@ -1,543 +1,2208 @@
+  identifiers; instead, they may choose to mark an identifier as "deprecated".
 
-The canonical source for the names and semantics of each of the supported
-metadata fields is the Core Metadata Specification [#cms]_ document.
+- The ``LicenseRef-Public-Domain`` and ``LicenseRef-Proprietary`` strings, to
+  identify licenses that are not included in the SPDX license list.
 
-The details of the updates considered to the Core Metadata Specification [#cms]_
-document as part of this PEP are described here and will be added to the
-canonical source once this PEP is approved.
+When processing the ``License-Expression`` field to determine if it contains
+a valid license expression, build and publishing tools:
 
+- SHOULD halt execution and raise an error if:
 
-Added in Version 2.2
---------------------
+  - The field does not contain a valid license expression
 
-License-File (multiple use)
-:::::::::::::::::::::::::::
+  - One or more license identifiers are not valid (as defined above)
 
-The License-File is a string that is a path, relative to``.dist-info``, to a
-license file. The license file content MUST be UTF-8 encoded text.
+- SHOULD report an informational warning, and publishing tools MAY raise an
+  error, if one or more license identifiers have been marked as deprecated in
+  the `SPDX License List <#spdxlist_>`_.
 
-Build tools SHOULD honor this field and include the corresponding license
-file(s) in the built package.
+- MUST store a case-normalized version of the ``License-Expression`` field
+  using the reference case for each SPDX license identifier and
+  uppercase for the ``AND``, ``OR`` and ``WITH`` keywords.
 
+- SHOULD report an informational warning, and MAY raise an error if
+  the normalization process results in changes to the
+  ``License-Expression`` field contents.
 
-Changed in Version 2.2
-----------------------
+For all newly-upload distributions that include a
+``License-Expression`` field, the `Python Package Index (PyPI) <#pypi_>`_ MUST
+validate that it contains a valid, case-normalized license expression with
+valid identifiers (as defined here) and MUST reject uploads that do not.
+PyPI MAY reject an upload for using a deprecated license identifier,
+so long as it was deprecated as of the above-mentioned SPDX License List
+version.
 
-License (optional)
-::::::::::::::::::
 
-Text indicating the license covering the distribution. This text can be either a
-valid license expression as defined here or any free text.
+Add ``License-File`` field
+''''''''''''''''''''''''''
 
-Publishing tools SHOULD issue an informational warning if this field is empty,
-missing, or is not a valid license expression as defined here. Build tools MAY
-issue a similar warning.
+The ``License-File`` optional field is specified to contain the string
+representation of the path to a license-related file, relative to the
+root license directory. It is a multi-use field that may appear zero or
```

Comment:
Ah, you've actually caught an editing mistake (originally, the sentence defining it was above this one). I was considering adding it to the Terminology section, but held off since it was only really directly referenced in a few places and didn't want to make the latter too long. I went go ahead with that, thanks.